test/OpenIddict.Server.AspNetCore.IntegrationTests/OpenIddictServerAspNetCoreIntegrationTestServer.cs

```diff
@@ -57,8 +57,10 @@ ValueTask DisposeAsync()
 
 #if SUPPORTS_GENERIC_HOST
             // Stop and dispose of the underlying generic host.
-            await Host.StopAsync();
-            Host.Dispose();
+            if (Host is object) {
```

Comment:
Very unsure about this one 🤔

Reply:
The problem is that the first ctor (https://github.com/openiddict/openiddict-core/blob/ba85a4b618f5e6065ca6310881af6b6c0fff04e3/test/OpenIddict.Server.AspNetCore.IntegrationTests/OpenIddictServerAspNetCoreIntegrationTestServer.cs#L20) does not initialize the `Host` property; therefore it must be marked as nullable.

Reply:
Oh, then we should rework the definition a bit to only add this constructor when generic host is not supported. Something like that:

```csharp
#if SUPPORTS_GENERIC_HOST
        public OpenIddictServerAspNetCoreIntegrationTestServer(IHost host)
        {
            Host = host;
            Server = host.GetTestServer();
        }
#else
        public OpenIddictServerAspNetCoreIntegrationTestServer(TestServer server)
            => Server = server;
#endif
```